third_party/libunwindstack/Unwinder.cpp

```diff
@@ -121,8 +121,36 @@ static bool ShouldStop(const std::vector<std::string>* map_suffixes_to_ignore,
                    map_name.substr(pos + 1)) != map_suffixes_to_ignore->end();
 }
 
-void Unwinder::Unwind(const std::vector<std::string>* initial_map_names_to_skip,
-                      const std::vector<std::string>* map_suffixes_to_ignore) {
+static bool ShouldStop(
+    const std::map<std::string, std::map<uint64_t /*function_start*/, uint64_t /*size*/>>*
+        functions_to_stop,
+    const std::string& map_name, uint64_t rel_pc) {
+  if (functions_to_stop == nullptr) {
+    return false;
+  }
+  auto functions_it = functions_to_stop->find(map_name);
+  if (functions_it == functions_to_stop->end()) {
+    return false;
+  }
+
+  auto function_it = functions_it->second.upper_bound(rel_pc);
+  if (function_it == functions_it->second.begin()) {
+    return false;
+  }
+
+  --function_it;
+
+  uint64_t function_start = function_it->first;
+  CHECK(function_start <= rel_pc);
+  uint64_t size = function_it->second;
+  return (rel_pc < function_start + size);
+}
+
+void Unwinder::Unwind(
+    const std::vector<std::string>* initial_map_names_to_skip,
+    const std::vector<std::string>* map_suffixes_to_ignore,
+    const std::map<std::string, std::map<uint64_t /*function_start*/, uint64_t /*size*/>>*
```

Comment:
So the key of the outer map is the module name? It would be useful to specify that too. Or to rename the parameter accordingly.

Reply:
Done.